File: AtCoder/EducationDP/Frog2/main.py

n, k = map(int, input().split())
h = list(map(int, input().split()))
dp = [100000000 for _ in range(n)]
dp[0] = 0
for i in range(n - 1):
    for j in range(i + 1, min(i + k + 1, n)):
        dp[j] = min(dp[j], dp[i] + abs(h[i] - h[j]))

print(dp[n - 1])

File: AtCoder/EducationDP/Frog2/main.cpp
#include <bits/stdc++.h>
#include <iostream>
#define ll long long
using namespace std;

ll dp[100001];

int main()
{
    int n, k;
    vector<int> heights;
    cin >> n >> k;
    
    for (int i = 0; i < n; i++)
    {
        int x;
        cin >> x;
        heights.push_back(x);
        dp[i] = INT_MAX;
    }

    dp[0] = 0;
    for (int i = 0; i < n; i++)
    {
        for (int j = i + 1; j < min(i + k + 1, n); j++)
        {
            dp[j] = min(dp[j], dp[i] + abs(heights[i] - heights[j]));
        }
    }

    cout << dp[n - 1] << endl;

}








